src/components/General/RegistrationModal.jsx

```diff
@@ -0,0 +1,244 @@
+import {
+  userRegister,
+  userSignIn,
+  getUserInfo,
+} from "../../store/user/services";
+import { useDispatch } from "react-redux";
+import { useHistory } from "react-router-dom";
+import { saveUser } from "../../store/selectUserReducer";
+import React, { useState } from "react";
+
+function RegistrationModal() {
+  const [FirstName, setFirstName] = useState("");
+  const [LastName, setLastName] = useState("");
+  const [DNIType, setDNIType] = useState("");
+  const [DNINumber, setDNINumber] = useState("");
+  const [Email, setEmail] = useState("");
+  const [Address, setAddress] = useState("");
+  const [Neighborhood, setNeighborhood] = useState("");
+  const [Phone, setPhone] = useState("");
+  const [Height, setHeight] = useState("");
+  const [Weight, setWeight] = useState("");
+  const [Birthday, setBirthday] = useState("");
+  const [Password, setPassword] = useState("");
+
+  const history = useHistory();
+  const dispatch = useDispatch();
+
+  const handleRegister = () => {
+    userRegister(
+      FirstName,
+      LastName,
+      DNIType,
+      DNINumber,
+      Email,
+      Address,
+      Neighborhood,
+      Phone,
+      Height,
+      Weight,
+      Birthday,
+      Password
+    ).then((resUserRegister) => {
+      userSignIn(Email, Password).then((resUserSignIn) => {
+        const { data: dataUserSignIn } = resUserSignIn;
+        if (dataUserSignIn.token) {
+          localStorage.setItem("token", dataUserSignIn.token);
+          getUserInfo(dataUserSignIn.token).then((resGetUserInfo) => {
```

Comment:
Alguna razón para hacer estas 3 peticiones? No es necesario registrarse, inmediatamente hacer login, e inmediatamente obtener la información del usuario. En el BE se puede definir cualquier propiedad en el response, no se limiten al ejemplo de la clase. Por ejemplo en el register puedo responder con el token (ya no necesita hacer login), en el register puedo adicionalmente responder con la información del usuario que se acaba de registrar o de loggear y así no necesito hacer la petición adicional de getUserInfo